### ISSUE TITLE
Java Edition 1.20.5

### ISSUE THREAD

**Issue by nahkd123** [nahkd123/stonks#48]
1.20.5 is currently in snapshot phase. I expect sgui to break because there are changes to networking code.

- [x] Upstream: Java Edition 1.20.5
- [x] Upstream: Fabric Loader for 1.20.5
- [x] Upstream: Fabric API for 1.20.5
- [x] Upstream: Patbox's sgui for 1.20.5
- Update Stonks to 1.20.5
  + [x] Deprecate Nahara Toolkit Tasks
  + [x] Remove Nahara Modkit Annotations and move back to manual JSON editing
  + [x] Update default configuration file to use components
  + [x] Write changelogs
  + [x] Test

**Comment by nahkd123** [comment]
1.20.5 is released!

**Comment by nahkd123** [comment]
I decided to take this opportunity to deprecate Nahara Toolkit Tasks in favor of Java Future API. Everything will be broken anyways.

**Comment by nahkd123** [comment]
```
java.lang.RuntimeException: Error creating Mixin config fabric-game-rule-api-v1.mixins.json for mod fabric-game-rule-api-v1
	at net.fabricmc.loader.impl.launch.FabricMixinBootstrap.init(FabricMixinBootstrap.java:96) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:151) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.10.jar:?]
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86) ~[?:?]
Caused by: java.lang.NoSuchFieldError: Class org.spongepowered.asm.mixin.MixinEnvironment$CompatibilityLevel does not have member field 'org.spongepowered.asm.mixin.MixinEnvironment$CompatibilityLevel JAVA_22'
	at net.fabricmc.loader.impl.launch.knot.MixinServiceKnot.getMaxCompatibilityLevel(MixinServiceKnot.java:227) ~[fabric-loader-0.15.10.jar:?]
	at org.spongepowered.asm.mixin.MixinEnvironment.setCompatibilityLevel(MixinEnvironment.java:1577) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.initCompatibilityLevel(MixinConfig.java:557) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.postInit(MixinConfig.java:502) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.onLoad(MixinConfig.java:430) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.create(MixinConfig.java:1293) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.Config.create(Config.java:148) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.Mixins.createConfiguration(Mixins.java:100) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.Mixins.addConfiguration(Mixins.java:87) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.FabricMixinBootstrap.init(FabricMixinBootstrap.java:94) ~[fabric-loader-0.15.10.jar:?]
	... 4 more
```

Am I doing something wrong?

**Comment by nahkd123** [comment]
Fabric Loader tries to use JAVA_22 for max compat level, but somehow Gradle resolves the Mixin dependency to version 0.12.5, which limits at JAVA_18

**Comment by nahkd123** [comment]
Turns out that Mixins 0.12.5 is coming indirectly from Common Economy API. Solved by excluding `net.fabricmc` from the dependency.
